Index: texworks-0.6.2/trans/TeXworks_trans.pro
===================================================================
--- texworks-0.6.2.orig/trans/TeXworks_trans.pro
+++ texworks-0.6.2/trans/TeXworks_trans.pro
@@ -31,9 +31,7 @@ SOURCES = \
   "../src/TeXDocument.cpp" \
   "../src/TeXHighlighter.cpp" \
   "../src/TemplateDialog.cpp" \
-  "../src/main.cpp" \
-  "../src/synctex_parser.c" \
-  "../src/synctex_parser_utils.c"
+  "../src/main.cpp"
 
 HEADERS = \
   "../src/ClickableLabel.h" \
@@ -65,9 +63,7 @@ HEADERS = \
   "../src/TeXDocks.h" \
   "../src/TeXDocument.h" \
   "../src/TeXHighlighter.h" \
-  "../src/TemplateDialog.h" \
-  "../src/synctex_parser.h" \
-  "../src/synctex_parser_utils.h"
+  "../src/TemplateDialog.h"
 
 FORMS = \
   "../src/ConfirmDelete.ui" \
